pspy/so_cov.py

```diff
@@ -672,14 +691,99 @@ def covariance_element_beam(id_element,
     return analytic_cov_from_beam
 
 
+def block_diagonal_mult(nblocks, slices, mbb_inv_ab_list, mbb_inv_cd_list, analytic_cov):
+
+    """Suggestion for adrien to do an operation of the type A M B, where A and B are block diagonal matrix
+    Parameters
+    ----------
+    nblocks: integer
+        the number of block of the block diagonal matrix
+    slces: dict of tuple of integer
```

Comment:
it is also not clear why `nblocks` needs to be passed. I understand that it directly comes from the `extract_mbb_list` but `nblocks` is actually `len(slices)` (even if `slices` is a `dict` which is a bit misleading since it is indexed over integer, the `len` function still works). We can imagine stupid things done by user such as passing a slices with less entries than `nblocks` values so I'd removed the `nblocks` parameters